lang/cs/Org.Apache.REEF.IMRU/API/UpdateResult.cs

```diff
@@ -50,22 +50,34 @@ internal bool IsNotDone
             get { return !_done; }
         }
 
-        internal bool HasMapInput
+        /// <summary>
+        /// Indicates if the UpdateResult has map input data
+        /// </summary>
+        public bool HasMapInput
```

Comment:
I understand  what you are saying public class doesn't means all the properties are public. I also agree in general we try not to expose method to public so that leaving room for us to change later. 

In this particular case, UpdateResult  is used in BroadcastSenderReduceReceiverUpdateFunction, an existing class in IMRU example, not just BroadcastSenderReduceReceiverUpdateFunctionFT. It is the object that is created in BroadcastSenderReduceReceiverUpdateFunction and returned to the TaskHost as intermediate data. As BroadcastSenderReduceReceiverUpdateFunction is user code (here is in IMRU example), so UpdateResult object is essentially created by user. Those properties, not even methods which we may change later. are to indicates what kind of result it contains. I don't see any issues to expose those properties in this class.